apps/examples/ble_perfs/ble_perfs_main.c

```diff
@@ -0,0 +1,452 @@
+		.attr_handle = BLE_STATE_MANAGER_RMC_HANDLE_KEY_CCCD, .cb = utc_cb_desc_b_1, .arg = "desc_1"},
+
+	{.type = BLE_SERVER_GATT_SERVICE, .uuid = {0xAD,0xB6,0x6E,0x45,0xA7,0x68,0x9D,0x8D,0x9A,0x40,0x17,0x2B,0xE9,0xCB,0xF2,0x13}, .uuid_length = 16,
+		.attr_handle = BLE_APP_HANDLE_SERVICE_1,},
+
+	{.type = BLE_SERVER_GATT_CHARACT, .uuid = {0x04,0xC7,0xAA,0xE7,0xF8,0x9A,0xCB,0x88,0x43,0x4C,0x44,0xCF,0x0D,0x5B,0xDA,0xF2}, .uuid_length = 16,
+		.property =  BLE_ATTR_PROP_RWN|BLE_ATTR_PROP_WRITE_NO_RSP, .permission = BLE_ATTR_PERM_R_PERMIT|BLE_ATTR_PERM_W_PERMIT,
+		.attr_handle = BLE_APP_HANDLE_CHAR_RMC_KEY, .cb = utc_cb_charact_a_1, .arg = "char_2"},
+
+	{.type = BLE_SERVER_GATT_DESC, .uuid = {0x02,0x29}, .uuid_length = 2,
+		.permission = BLE_ATTR_PERM_R_PERMIT|BLE_ATTR_PERM_W_PERMIT,
+		.attr_handle = BLE_APP_HANDLE_DESC_RMC_KEY, .cb = utc_cb_desc_b_1, .arg = "desc_2"},
+
+
+	{.type = BLE_SERVER_GATT_SERVICE, .uuid = {0xF4,0x7A,0x07,0x08,0xFD,0xC7,0x9D,0xB5,0xFF,0x4E,0x85,0xDE,0x48,0x80,0xFE,0xA2}, .uuid_length = 16,
+		.attr_handle = BLE_APP_HANDLE_SERVICE_2,},
+
+	{.type = BLE_SERVER_GATT_CHARACT, .uuid = {0x06,0xC7,0xAA,0xE7,0xF8,0x9A,0xCB,0x88,0x43,0x4C,0x44,0xCF,0x0D,0x5B,0xDA,0xBB}, .uuid_length = 16,
+		.property =  BLE_ATTR_PROP_RWN|BLE_ATTR_PROP_WRITE_NO_RSP, .permission = BLE_ATTR_PERM_R_PERMIT|BLE_ATTR_PERM_W_PERMIT,
+		.attr_handle = BLE_APP_HANDLE_CHAR_RMC_SYNC, .cb = ble_peri_cb_charact_rmc_sync, .arg = &g_perf_data},
+};
+
+static void ble_server_connected_cb(ble_conn_handle con_handle, ble_server_connection_type_e conn_type, uint8_t mac[BLE_BD_ADDR_MAX_LEN])
+{
+	char address_str[BLE_BD_ADDR_STR_LEN + 1] = {0,};
+
+	BLE_LOGD( "conn : %d / conn_type : %s(%d) \n", con_handle, get_conn_type_str(conn_type), conn_type);
+
+	addr_type_to_string(address_str, mac);
+	BLE_LOGD( "device address [%s] \n", address_str);
+
+	if (conn_type == BLE_SERVER_DISCONNECTED) {
+		perfs_ble_reset_data();
+	}
+
+	return;
+}
+
+static ble_server_init_config server_config = {
+	ble_server_connected_cb,
+	true,
+	gatt_profile, sizeof(gatt_profile) / sizeof(ble_server_gatt_t)};
+
+
+static void perfs_ble_usage(void)
+{
+	BLE_LOGD( "BLE Perfs Usage: \n\n" \
+		 " ble_perfs start \n" \
+		 " ble_perfs adv \n" \
+		 " ble_perfs stop \n" \
+		 " ble_perfs result \n");
+}
+
+static int perfs_ble_server_init(void)
+{
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	ble_result_e ble_ret = BLE_MANAGER_FAIL;
+
+	/*1. Enable WiFi
+	2. Enable BLE Manager
+	*/
+
+	res = wifi_manager_init(&g_wifi_callbacks);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		BLE_LOGD( "Fail to init wifi manager [%d] \n", res);
+		return -1;
+	}
+
+	usleep(100 * 1000);
+
+	ble_ret = ble_manager_init(&server_config);
+	if (ble_ret != BLE_MANAGER_SUCCESS) {
+		if (ble_ret == BLE_MANAGER_ALREADY_WORKING) {
+			BLE_LOGD( "ble init is already done \n");
+		} 
+		BLE_LOGD( "Fail to init ble manager \n");
+		return -1;
+	}
+
+	g_is_ble_serv_init = 1;
+	usleep(100 * 1000);
```

Comment:
May be we can add the sleep seconds to a #define 
#define USLEEP_USEC

Reply:
done